fuse/ipns/ipns_unix.go

```diff
@@ -370,13 +384,21 @@ func (dir *Directory) Mkdir(ctx context.Context, req *fuse.MkdirRequest) (fs.Nod
 
 func (fi *File) Open(ctx context.Context, req *fuse.OpenRequest, resp *fuse.OpenResponse) (fs.Handle, error) {
 	if req.Flags&fuse.OpenTruncate != 0 {
-		log.Warning("Need to truncate file!")
+		log.Info("Need to truncate file!")
 		err := fi.fi.Truncate(0)
 		if err != nil {
 			return nil, err
 		}
 	} else if req.Flags&fuse.OpenAppend != 0 {
-		log.Warning("Need to append to file!")
+		log.Info("Need to append to file!")
+
+		// seek(0) essentially resets the file object, this is required for appends to work
+		// properly
+		_, err := fi.fi.Seek(0, os.SEEK_SET)
```

Comment:
not sure about the underlying structure-- 
- why isn't this happening for all opens? 
- how come the same `fi *File` can be opened by multiple things and set the seek position to different spots. 

my impression is that an open call should allocate a handle/descriptor of some sort with its own seek offset.